pkg/reftracker/ref_tracker.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package reftracker
+
+import (
+	"fmt"
+	"strings"
+)
+
+var appExists struct{}
+
+type AppRefTracker struct {
+	refsToApps         map[string]map[string]struct{}
+	appsToUpdateStatus map[string]struct{}
```

Comment:
seems that this object is doing two fairly distinct things: keeping track of app refs and also keeping track if app should be forced to update. let's split them up into two separate objects.

Reply:
I am wondering if it actually would make sense to just have a single map and change `refToApps` to `map[string]map[string]bool`. The main reason for using `appsToUpdateStatus` was for supporting the following that is going to be removed:

```
// Leave as not updated if error occurs
// in case follow up reconcile request may
// address issue for App.
r.appRefTracker.MarkAppUpdated(existingApp.Name, existingApp.Namespace)
```

Reply:
Ended up going with your advice to split to separate struct.